mat/band.go

```diff
@@ -0,0 +1,147 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat
+
+import (
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+var (
+	bandDense *BandDense
+	_         Matrix  = bandDense
+	_         Band    = bandDense
+	_         RawBand = bandDense
+)
+
+// BandDense represents a band matrix in dense storage format.
+type BandDense struct {
+	mat blas64.Band
+}
+
+type Band interface {
```

Comment:
I think we want the interface to be `Banded` rather than `Band` to match `Triangular` and `Symmetric`. I'm not sure if we want the type to be `BandedDense` instead of `BandDense`